lisa/runners/legacy_runner.py

```diff
@@ -584,12 +584,15 @@ def _read_log(self) -> str:
         # detach the handle
         detached_handle = handle.Detach()
 
+        content = ""
         # get a file descriptor associated to the handle
-        file_descriptor = msvcrt.open_osfhandle(detached_handle, os.O_RDONLY)
+        if not TYPE_CHECKING:  # FIXME: if you have a better solution
+            # for mypy checks on Linux, change this
+            file_descriptor = msvcrt.open_osfhandle(detached_handle, os.O_RDONLY)
 
-        # open the file descriptor
-        with open(file_descriptor) as file:
-            content = file.read()
+            # open the file descriptor
+            with open(file_descriptor) as file:
+                content = file.read()
```

Comment:
Before return(), close the fd.
If no 'TYPE_CHECKING', then this function return empty string. Should we add any error or debug msg?

Reply:
Wrong, it will return "" *if* type-checking.

Reply:
I meant 'else' block.

Reply:
Let's see if we're on the same page here:
- content is initialized to empty
- if we're type-checking, we jump straight to returning that. there is no harm, since it's a static check analysis, there is code being actually run
- if we're *not* type-checking, then we have the old behavior and content is assigned something else, before we return
- the file descriptor is never leaked, as it's under a with statement

Anything I'm foreseeing, still?

Reply:
Understood. My comment is about basic programming rules in our hands. This runs against new stuff, called "testing", and our programming might be defensive. So I commented to do it rather than leaving in the VM side. Your PR is totally fine in the normal case, however our run-time is not always "normal". If you think my comment is worthless, then you can close it. No need to ping pong here.

Reply:
I see, you're afraid some bozo will set that value artificially. Fair. So any better idea on how to handle that failing on Linux? The stub seems to be there, but mypy does not like it...